xrpl/core/binarycodec/types/issue.py

```diff
@@ -87,7 +102,11 @@ def to_json(self: Self) -> Union[str, Dict[Any, Any]]:
         Returns:
             The JSON representation of an Issue.
         """
-        parser = BinaryParser(str(self))
+        # If the buffer is exactly 24 bytes, treat it as an MPT amount.
+        if len(self.buffer) == 24:
+            return {"mpt_issuance_id": self.to_hex().upper()}
```

Comment:
Can you add an implementation for the `MPTAmount.from_parser` method? I agree that the 24-byte length-check is sufficient for our purposes. 

However, if developers want to de-serialize a `tx_blob` which contains `MPTAmount` fields, then `BinaryParser().from_parser()` is a standard way to handle all the internal types. 

For instance, the below usage is useful.

```
mpt_amount = MPTAmount.from_parser(parser, length_hint=24).to_json()
```

Reply:
The other currency types don't implement it. So I'm sure developers won't be calling it directly since the conversion is already done when using this client library.

Reply:
@khancode I took a perfunctory glance at the type implementations. `Amount`, `Hash`, `PathSet`, `SerializedType`, `STArray`, `STObject` -- all of them implement it.

Even if a type doesn't implement the `from_parser` method, it will rely on its parent classes implementation.

Are there any types which do not have this method?

Reply:
The types I'm referring to that are within Issue model: `XRP`, `IssuedCurrency`.